docs/developers/tutorials/energy-dao.md

```diff
@@ -0,0 +1,878 @@
+
+This template contract splits the __Farm__ integration in 2 different files, for better readability. One with the actual user interactions (the endpoints), where all the custom computation are done, and another one with the more generic actions regarding the DEX farm contract integration and any other general functions needed on this part.
+
+[comment]: # (mx-context-auto)
+
+### Farm actions
+
+Here we have a code snippet, that does the actual interaction with the farm contract. We use a __farm_proxy__ imported from the DEX reference declared in the Cargo.toml file, proxy that receives the address of the corresponding farm. We then call the desired endpoint on the farm contract (in our case __enter_farm_endpoint__) using a multi_token_transfer of a __PaymentsVec__ received as an argument. Under the hood, this is a __ManagedVec__ of __ESDTTokenPayments__ usually consisting in two payments, the first one the being the position with which the user wants to enter the contract, and the second one the rest of the contract's aggregated position. Finally, the function returns an __EnterFarmResultType__ (type imported from the farm contract), which represents a __MultiValue2__ of __ESDTTokenPayments__, the first payment representing the new aggregated farm position, and the second one the boosted_rewards, if any.
+
+```rust
+    fn call_enter_farm(
+        &self,
+        farm_address: ManagedAddress,
+        farming_tokens: PaymentsVec<Self::Api>,
+    ) -> EnterFarmResultType<Self::Api> {
+        self.farm_proxy(farm_address)
+            .enter_farm_endpoint(OptionalValue::<ManagedAddress>::None)
+            .with_multi_token_transfer(farming_tokens)
+            .execute_on_dest_context()
+    }
+
+    #[proxy]
+    fn farm_proxy(&self, sc_address: ManagedAddress) -> farm_with_locked_rewards::Proxy<Self::Api>;
+```
+
+:::important
+A very important aspect here is that in order to work as intended, the EnergyDAO SC must be deployed on the same shard as the DEX, in order to use intrashard contract calls and have syncronous, realtime SC results from the xExchange contracts.
```

Comment:
to work with the current design. With AsyncV2 these kinds of contracts can be deployed in other shards, as the protocol will support multiple asyncCalls.

Reply:
Updated.